airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3NameTransformer.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.s3;
+
+import io.airbyte.integrations.destination.StandardNameTransformer;
+import java.text.Normalizer;
+import java.util.regex.Pattern;
+
+public class S3NameTransformer extends StandardNameTransformer {
+
+  // see https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
+  private static final String S3_SAFE_CHARACTERS = "\\p{Alnum}/!_.*')(";
+  private static final String S3_SPECIAL_CHARACTERS = "&$@=;:+,?-";
```

Comment:
related to https://github.com/airbytehq/airbyte/pull/11173/

This is a bit more general though